Add Hashicorp Vault KV storage

slowpokeeee opened the pull request:
This PR makes it possible to load settings from a Hashicorp Vault KV (Key-Value) Secret Engine.
* Uses the [hvac](https://github.com/hvac/hvac) module to access the API

There where a few parts which I don't know if there is better ways to do it, so please come with comments.
1. Naming. Hashicorp Vault comes with multiple Secret Enginges. For normal configs/secrets, KV's seems to be the most obvious one and probably the only one that will be used, so I didn't bother to name the class `HVACKVLoader`. Maybe HVAC is not even good, but `VaultLoader` might be misleading as well since there is multiple vault vendors our there.
2. Should an `is_authenticated()` be called after `client` has been created? I didn't do it since I thought it might be a bit overhead, but on the other hand we can catch an exception earlier if we do it.

@Stealthmate Sorry, since you talked about wanting to have Vault option, you might have planned to make it. Feel free to come with any suggestions. Also if you want me to put this to a separate branch so you can make changes as well, just say so!

Commit:
feat: Add Hashicorp Vault KV storage

Stealthmate commented:
Alright, I went through this as carefully as I can, and well, turns out Vault is a lot more complicated than I thought lol.

First off, to answer your questions:

1. I had to read about the concept of an engine in the first place, but if I understand correctly, secret engines are basically an abstraction for storing different _types_ of secrets. Moreover, most of the time they are meant to create automatic credentials. With that in mind, I think it would be better to have a different loader per engine type (since we _have_ to specify the engine when using the client).
2. Good question. I think it's better to do auth (or any kind of external request for that matter) inside `load` rather than the constructor. It is up to the library user to decide when/how often to call `load`, but we should give them the freedom to create the loader without causing any side effects (similar to how the YAML/JSON/TOML loaders preform file operations in `load` as well).

Also, I think for other Vault engines, we should probably have some kind of way to specify how to map the Vault values to a dict inside Python. For KV engine it makes sense to just return the entire JSON content I guess. But for other engines that might not work. For example, I don't really know how AWS IAM User credentials would be stored/returned from the client: https://developer.hashicorp.com/vault/docs/secrets/aws

Then again, we can merge this as a PR that simply adds support for HVAC KV. Other engines can get their own PRs.

Commit:
chore: move init of client to load

slowpokeeee commented:
Thanks for the review. I renamed the function to contain the KV Secret Engine. Since the client is basically only used in the `load()` function, I moved the initialization there as well. I don't believe the creation of the client is any overhead, but for not since we don't use it anyway it's maybe better to have it in `load()`. 

If there is anything else I consider feel free to point it out.
